From 2e03a5b3ffc490028a2dbb8042d5036157464fe8 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Fri, 4 Feb 2022 15:04:41 -0800
Subject: Fix go test tools (#37177)

TestMain may cause `-list` to print out anything. So make sure we just
capture what we want
---
 tools/go-ordered-test | 2 +-
 tools/go-stress-test  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/go-ordered-test b/tools/go-ordered-test
index a5bc04ca4d..f86c4e74b1 100755
--- a/tools/go-ordered-test
+++ b/tools/go-ordered-test
@@ -44,7 +44,7 @@ RESULTS=/tmp/test-results"$(dirname ${binary})"
 mkdir -p "${RESULTS}"
 code=0
 
-for testname in $("${binary}" -test.list "${RUN}"); do
+for testname in $("${binary}" -test.list "${RUN}" | grep '^Test'); do
   "${binary}" -test.run '^'"${testname}"'$' -test.count "${COUNT}" -test.v &> "${RESULTS}/${testname}"
   # shellcheck disable=SC2181
   if [[ $? != 0 ]]; then
diff --git a/tools/go-stress-test b/tools/go-stress-test
index 79b336b543..1b9d3fc850 100755
--- a/tools/go-stress-test
+++ b/tools/go-stress-test
@@ -53,7 +53,7 @@ RESULTS=/tmp/test-results"$(dirname ${binary})"
 mkdir -p "${RESULTS}"
 code=0
 
-for testname in $("${binary}" -test.list "${RUN}"); do
+for testname in $("${binary}" -test.list "${RUN}" | grep '^Test'); do
   stress -f --max-time "${TIME:-10s}" --max-runs "${RUNS:-1000}" "${binary}" -test.run '^'"${testname}"'$' -test.count "${COUNT}" -test.v &> "${RESULTS}/${testname}"
   # shellcheck disable=SC2181
   if [[ $? != 0 ]]; then
-- 
2.35.3

